### PR TITLE
Get all attributes

### PR DESCRIPTION
**Pull request by dmullen17**
Function to pull all attribute metadata out of an EML in a list.  Has the added functionality of being able to enter a URL and node to download metadata from.  
Feel like a quick code review @isteves @maier-m ?

### CODE REVIEW

**Comment by isteves** [comment]
@rrcarlson try code reviewing this! Let us know if you're confused on how and we can show you
@dmullen17 I can also look over it if Mitchell doesn't get to it first

File: R/get_all_attributes.R [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    if (!is.null(eml)) {
+        stopifnot(isS4(eml))
+    }
+
+    # If node is specified run checks, otherwise set node = ADC production
+    if (!is.null(node)) {
+        stopifnot(methods::is(node, "MNode") || is(node, "CNode"))
+    } else {
+        cn <- dataone::CNode('PROD')
+        node <- dataone::getMNode(cn,'urn:node:ARCTIC')
```
Comment (dmullen17) [inline R/get_all_attributes.R]: take this out

File: R/get_all_attributes.R [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        eml <- EML::read_eml(rawToChar(dataone::getObject(node, pid)))
+    }
+
+    n <- length(eml@dataset@dataTable)
+    if (n == 0){
+        stop("EML does not contain attribute metadata")
+    }
+
+    results <- lapply(eml@dataset@dataTable, function(dataTable) {
+        att <- EML::eml_get(dataTable, "attributeList", "entityName")
```
Comment (dmullen17) [inline R/get_all_attributes.R]: take "entityName" out

File: R/get_all_attributes.R [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        pid <- unlist(strsplit(url, "view/"))[[2]]
+        eml <- EML::read_eml(rawToChar(dataone::getObject(node, pid)))
+    }
+
+    n <- length(eml@dataset@dataTable)
+    if (n == 0){
+        stop("EML does not contain attribute metadata")
+    }
+
+    results <- lapply(eml@dataset@dataTable, function(dataTable) {
```
Comment (isteves) [inline R/get_all_attributes.R]: attributes <- eml_get(eml, "attributeList")

File: R/get_all_attributes.R [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        att <- EML::eml_get(dataTable, "attributeList", "entityName")
+    })
+
+    names(results) <- lapply(eml@dataset@dataTable, function(dataTable) {
```
Comment (dmullen17) [inline R/get_all_attributes.R]: add extracting attributes from otherEntity

File: R/get_all_attributes.R [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        EML::eml_get(dataTable, "entityName")
```
Comment (isteves) [inline R/get_all_attributes.R]: do this on attributes list

**Pull request merged**